added checklists about builder and files (#6)
mate-academy/jv-program-common-mistakes: java-core/builder-file/read-from-file.md

### Common mistakes

#### Remember about informative name of the variables.
```
Bad naming:
BufferedReader br = new BufferedReader(new FileReader(fileName))
```  
```
Good naming: 
BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))
```  

### Constants.

> Some strange strings or numbers in the code better make as constants. 

Reviewer: ```suggestion
If you have strange strings or numbers in the code it's better to declare them as constants.
```

> The name of this constant should be the information about what this objects for.

Reviewer: ```suggestion
The name of the constant should display this object's purpose.
```

```
Bad practice:
    public boolean startWithLetter(String word) {
        return word.startsWith("d"); // why do we use 'd' here???
    }
```
```
Good practice:
    private static final String SPECIFIED_CHARACTER = "d";
    
    public boolean startWithLetter(String word) {
        return word.startsWith(SPECIFIED_CHARACTER); 
    }
```
[Correct constant names](https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names)

### Close all resources.
Connections, streams, files, and other classes that implement the `Closeable` or `AutoCloseable` interface, 

> needs to be closed after use. Further, that close must be made in a `finally` block. 

Reviewer: ```suggestion
needs to be closed after use. Furthermore, that close should be done in a `finally` block. 
```
